tests/5.0/loop/test_loop_lastprivate.F90

```diff
@@ -0,0 +1,107 @@
+!===--- test_loop_lastprivate.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test uses the lastprivate clause with a loop directive. According to
+! specification, the list items on a lastprivate clause in this context may
+! only contain loop iteration variables of loops associated with the loop
+! directive. This test checks that the loop iteration variables associated
+! with a loop directive and a loop directive with collapse(2) have valid
+! values after the parallel region containing the loop.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define NSIZE 1024
+#define NSIZE2 512
+
+PROGRAM test_loop_lastprivate
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_VERBOSE(test_one_loop_level() .ne. 0)
+  OMPVV_TEST_VERBOSE(test_two_loop_levels() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION test_one_loop_level()
+    INTEGER,DIMENSION(NSIZE):: a, b
+    INTEGER:: errors, lp_errors, x
+    CHARACTER(len=400) :: msgHelper
+
+    errors = 0
+    lp_errors = 0
+    x = 0
+
+    DO x = 1, NSIZE
+       a(x) = 1
+       b(x) = x
+    END DO
+
+    !$omp parallel num_threads(OMPVV_NUM_THREADS_HOST)
+    !$omp loop lastprivate(x)
+    DO x = 1, NSIZE
+      a(x) = a(x) + b(x)
+    END DO
+    !$omp end loop
+    !$omp end parallel
+
+    OMPVV_TEST_AND_SET_VERBOSE(lp_errors, x .ne. NSIZE)
+    WRITE(msgHelper, *) "Loop iteration variable in loop construct ended with invalid value."
+    OMPVV_ERROR_IF(x .ne. NSIZE, msgHelper)
```

Comment:
Compiles and runs with GCC mainline (13; should likewise run with released GCC 12). However:
* `lastprivate` value is NSIZE + 1 after the loop.  (This matches Fortran's "11.1.7.4.3 The execution cycle" and the OpenMP rules).
* The test in line line 53 and 55 is effectively the same. Is this extra check needed? Additionally, instead of an internal write, an assignment would do in line 54 as it is currently just a string. Or  it is changed to `"with invalid value: ",x` then it has an added value (both direct and idiomatic meaning).

Likewise below – and also for `y` and `NSIZE2` .